rulewerk-core/src/main/java/org/semanticweb/rulewerk/core/reasoner/implementation/TermToVLogConverter.java

```diff
@@ -86,6 +86,19 @@ public static String getVLogNameForConstant(Constant constant) {
 		}
 	}
 
+	/**
+	 * Converts the given named null to the name of a (skolem) constant in VLog.
+	 *
+	 * @fixme This skolemisation approach might lead to constants that clash with
+	 *        existing constant names.
+	 *
+	 * @param named null
+	 * @return VLog constant string
+	 */
+	public static String getVLogNameForNamedNull(NamedNull namedNull) {
```

Comment:
A better approach would be to use `org.semanticweb.rulewerk.core.reasoner.implementation.Skolemization#skolemizeNamedNull` here (the converter can have a static field that holds a `Skolemization` instance with its own namespace), and to provide an overload for `org.semanticweb.rulewerk.core.model.implementation.RenamedNamedNull` to avoid renaming already renamed named nulls again.

Reply:
Interesting. I did not know that you can overload with arguments that are in a subclass relationship. That did not always work in Java, or did it?